### PR TITLE
Remove use of specific atom class and use generic atom instead

### PR DESCRIPTION
**Pull request by talekhinezh**
* RE should only be responsible for RE level logic so use interface rather than class to retrieve required RE data

### CODE REVIEW

File: src/main/java/com/radixdlt/engine/RadixEngineAtom.java [hunk 1]
```diff
@@ -18,7 +18,23 @@
 package com.radixdlt.engine;
 
 import com.radixdlt.constraintmachine.CMInstruction;
+import com.radixdlt.identifiers.AID;
 
+/**
+ * Radix Engine interface for retrieving a sequence of instruction to
```
Comment (flotothemoon) [inline src/main/java/com/radixdlt/engine/RadixEngineAtom.java]: instructions

**Review by flotothemoon — APPROVED** [review]
Fine by me

File: src/test/java/com/radixdlt/engine/RadixEngineTest.java [hunk 2]
```diff
@@ -54,14 +54,15 @@ public void when_validating_an_atom_with_particle_which_conflicts_with_virtual_s
 		when(constraintMachine.validate(any())).thenReturn(Optional.empty());
 		EngineStore engineStore = mock(EngineStore.class);
```
Comment (clstrfsck) [inline src/test/java/com/radixdlt/engine/RadixEngineTest.java]: EngineStore missing a type argument, which causes Eclipse compiler to type-fail on RadixEngine construction below.

File: src/main/java/com/radixdlt/engine/RadixEngine.java [hunk 3]
```diff
@@ -45,84 +42,62 @@
 /**
  * Top Level Class for the Radix Engine, a real-time, shardable, distributed state machine.
  */
-public final class RadixEngine {
+public final class RadixEngine<T extends RadixEngineAtom> {
 
 	private static final Logger log = LogManager.getLogger(RadixEngine.class);
```
Comment (clstrfsck) [inline src/main/java/com/radixdlt/engine/RadixEngine.java]: No longer used -> remove.

**Review by clstrfsck — CHANGES_REQUESTED** [review]
The missing type argument needs fixing in RadixEngineTest, and would prefer to see unused logger removed also.

**Review by clstrfsck — APPROVED** [review]

**Pull request merged**